### ISSUE TITLE
Rule Change: `no-useless-catch`: don't consider it useless if there's a `finally` clause

### ISSUE THREAD

**Issue by Prinzhorn** [eslint/eslint#16396]
### What rule do you want to change?

no-useless-catch

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
async function trackedFetch(...args) {
  actions.inc();

  try {
    return await fetch(...args);
  } catch (err) {
    throw err;
  } finally {
    actions.dec();
  }
}
```

### What does the rule currently do for this code?

> Unnecessary catch clause.  ([no-useless-catch](https://eslint.org/docs/rules/no-useless-catch))

### What will the rule do after it's changed?

:heavy_check_mark: 

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

Maybe I'm missing something, it's late. I'm wrapping `fetch` calls in `trackedFetch` function so that I can keep count of the number of active requests (yes, axios interceptors exist, I know).

In this case `throw err` is not pointless. I need the try/catch block so that I can decrement the counter in `finally`. For the caller of `trackedFetch` it should still behave like `fetch` and throws error as usual, that's why I need to rethrow the error.

**Comment by ljharb** [comment]
You don't need the `catch` at all - this is valid code:
```js
  try {
    return await fetch(...args);
  } finally {
    actions.dec();
  }
```
in other words, the rule is correct, this catch is useless, and you should remove it.

**Comment by Prinzhorn** [comment]
Well, I knew I was missing something. Thank you very much!
